### PR TITLE
expanded error type with tests

### PR DESCRIPTION
**Pull request by kujenga**
Adds handling for the inner tag format [1] and HTTP headers to facilitate improved handling of rate limiting [2]

[1] https://www.dropbox.com/developers/reference/migration-guide#error-handling
[2] bottom of: https://www.dropbox.com/developers-v1/core/bestpractices

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -1,13 +1,45 @@
...
+	Err        map[string]interface{} `json:"error"`
 }
 
 // Error string.
 func (e *Error) Error() string {
 	return e.Summary
 }
+
+// Tag returns the inner tag for the error
+func (e *Error) Tag() (tag string, value string) {
```
Comment (kevindoyle) [inline error.go]: Is this preferable over `(tag, value string)`?

**Comment by kevindoyle** [comment]
Having the header in the Error will be nice! One small style comment. This looks great!

File: error.go (hunk 1 above)
Comment (kujenga) [inline error.go]: amended!

**Comment by kevindoyle** [comment]
Awesome! :+1:

**Pull request merged**